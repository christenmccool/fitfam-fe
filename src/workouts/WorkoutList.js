import React, {useState, useEffect} from 'react';

import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

import FitFamApi from '../api/api';
import WorkoutCardList from './WorkoutCardList';

/** Shows list of featured workouts for a given date
 * 
 * On mount, loads workouts from API
 * 
 * WorkoutList -> WorkoutCardList -> WorkoutCard 
 * 
 */
const WorkoutList = ({ date }) => {
  const [workouts, setWorkouts] = useState([]);

  useEffect(() => {
    async function getFeaturedWorkouts() {
      try {
        const dailyworkouts = await FitFamApi.getFeaturedWorkouts(date);
        setWorkouts(dailyworkouts);
      } catch (err) {
        console.log(err);
      }
    }
    getFeaturedWorkouts();
  }, []);

  if (!workouts) return <div>Loading</div>;

  return (
    <Box mt={2}>
      <Typography variant="h6" align="center" color="text.secondary" gutterBottom>
        {workouts.length===1 ? "Featured Workout" : "Featured Workouts"}
      </Typography>
      <WorkoutCardList
        workouts={workouts}
        cardMaxHeight={1000}
      />
    </Box>
  )
}

export default WorkoutList;