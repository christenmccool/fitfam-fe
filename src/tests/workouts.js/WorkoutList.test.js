import React from "react";
import { render, act } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import WorkoutList from "../../workouts/WorkoutList";

import FitFamApi from '../../api/api';
jest.mock('../../api/api');

beforeEach(() => {
  FitFamApi.getFeaturedWorkouts.mockResolvedValue([
      {
        id: 1, 
        name:"Test Workout Name 1", 
        description:"Do this hard thing", 
        category: "featured", 
        scoreType: "Reps", 
        createDate: "20220322",
        featuredDate: "20220322",
        createBy: null
      },
      {
        id: 2, 
        name:"Test Workout Name 2", 
        description:"Do this other hard thing", 
        category: "featured", 
        scoreType: "Reps", 
        createDate: "20220322",
        featuredDate: "20220322",
        createBy: null
      }
  ])
})

it('renders without crashing', async function () {
  await act(async () => {
    render(
      <MemoryRouter>
        <WorkoutList />
      </MemoryRouter>
    );
  });
});


it("matches snapshot", async function () {
  await act(async () => {
  const { asFragment } = render(
      <MemoryRouter>
        <WorkoutList />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});


it("renders list of workouts", async function () {
  const { findByText, getByText } = render(
    <MemoryRouter>
      <WorkoutList />
    </MemoryRouter>
  );
  expect(await findByText("Featured Workout", {exact: false})).toBeInTheDocument();
  expect(getByText("Test Workout Name 1")).toBeInTheDocument();
  expect(getByText("Test Workout Name 2")).toBeInTheDocument();
});





