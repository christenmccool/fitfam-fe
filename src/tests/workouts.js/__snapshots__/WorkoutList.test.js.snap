// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-19rm58x"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Loading
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
